eip_operator/src/controller/node.rs

```diff
@@ -157,3 +186,139 @@ impl k8s_controller::Context for Context {
+#[instrument(skip(api), err)]
+async fn add_gateway_status_label(
+    api: &Api<Node>,
+    name: &str,
+    status: &str,
+) -> Result<Node, kube::Error> {
+    info!(
+        "Adding gateway status label {} key {} to node {}",
+        crate::EGRESS_NODE_STATUS_LABEL,
+        status,
+        name
+    );
+    let patch = serde_json::json!({
+        "apiVersion": "v1",
+        "kind": "Node",
+        "metadata": {
+            "labels": {
+                // Ensure the status is lowercase to match conventions
+                crate::EGRESS_NODE_STATUS_LABEL: status.to_lowercase(),
+            }
+        }
+    });
+    let patch = Patch::Apply(&patch);
+    let params = PatchParams::apply(crate::FIELD_MANAGER);
+    api.patch(name, &params, &patch).await
+}
+
+/// Retrieve all egress nodes in the cluster.
+async fn get_egress_nodes(api: &Api<Node>) -> Result<Vec<Node>, kube::Error> {
+    let params = ListParams::default().labels(
+        format!(
+            "{}={}",
+            EGRESS_GATEWAY_NODE_SELECTOR_LABEL_KEY, EGRESS_GATEWAY_NODE_SELECTOR_LABEL_VALUE
+        )
+        .as_str(),
+    );
+
+    match api.list(&params).await {
+        Ok(node_list) => Ok(node_list.items),
+        Err(e) => Err(e),
+    }
+}
+
+/// Get Ready status from the node status field.
+async fn get_ready_status_from_node(node: &Node) -> Option<String> {
+    node.status
+        .as_ref()?
+        .conditions
+        .as_ref()?
+        .iter()
+        .find(|c| c.type_ == "Ready")
+        .map(|condition| condition.status.clone())
+}
+
+/// Retrieve node names and ready status given a list of nodes.
+async fn get_nodes_ready_status(node_list: Vec<Node>) -> Result<BTreeMap<String, String>, Error> {
+    let mut node_ready_status_map = BTreeMap::new();
+
+    for node in node_list {
+        if let Some(ref node_name) = node.metadata.name {
+            let ready_status = get_ready_status_from_node(&node)
+                .await
+                .ok_or(Error::MissingNodeReadyCondition)?;
+
+            node_ready_status_map.insert(node_name.to_string(), ready_status);
+        }
+    }
+
+    Ok(node_ready_status_map)
+}
+
+/// Find old nodes in a ready state of `Unknown`.
+/// Update their status label if another node is available.
+/// Note: Egress traffic will be immediately dropped once the egress status label is changed away from "true"
+#[instrument(skip(), err)]
+async fn cleanup_old_egress_nodes(eip_api: &Api<Eip>, node_api: &Api<Node>) -> Result<(), Error> {
+    // Gather a list of egress nodes and EIPs to check for potential cleanup.
+    let node_list = get_egress_nodes(&Api::all(node_api.clone().into())).await?;
+    match node_list.len() {
+        // Exit early and avoid extra network calls.
+        len if len < 2 => {
+            info!("Less than two egress nodes found, no cleanup is possible.");
+            return Ok(());
+        }
+        // At least two nodes found, continue with cleanup.
+        _ => {
```

Comment:
this whole match is just a simple `if node_list.len() < 2`

we can do an early exit to make the code a little more simple:

```rust
if node_list.len() < 2 {
  info!("...");
  return Ok(());
}

let node_names_and_status: BTreeMap<String, String> =
                get_nodes_ready_status(node_list).await?; 
...
```

Reply:
Much simpler. Fixed in [commit](https://github.com/MaterializeInc/k8s-eip-operator/pull/474/commits/1b1d20403323796aaad28a08d04affd3bcf0cada)